src/Auth0Client.ts

```diff
@@ -638,32 +638,36 @@ export default class Auth0Client {
       }
     }
 
-    try {
-      await lock.acquireLock(GET_TOKEN_SILENTLY_LOCK_KEY, 5000);
-      // Check the cache a second time, because it may have been populated
-      // by a previous call while this call was waiting to acquire the lock.
-      if (!ignoreCache) {
-        let accessToken = getAccessTokenFromCache();
-        if (accessToken) {
-          return accessToken;
-        }
-      }
+    while (true) {
```

Comment:
Could you explain please why the whole block was wrapped into `while (true) {}`?

Reply:
We are acquiring a lock using a timeout of `x`. Whenever the timeout is hit, the library resolves with false, indicating acquiring a lock was unsuccessful.

By wrapping it in a while(true), we are retrying until a point where the lock can be acquired. 

Previously, we werent taking the return value of acquireLock into account, resulting in releasing all queued calls when the timeout was hit. This while loop solves that in such a way that it will not release the queued calls but will keep them queued untill a lock was acquired.

I do think we might need to add an escaping mechanism, to avoid the possibility for a never ending loop, hanging and crashing the user's browser.

Reply:
We can add a max number of retries: If we have retried for > 10 times (as an example), we can throw an appropriate error to the user. A good value for the number of retries can be determined based on the average number of retries it takes given ~15 tabs open.. or something along those lines.

What do you think of this approach?

Reply:
> it will not release the queued calls but will keep them queued until an actual lock was acquired.

> We can add a max number of retries: If we have retried for > 10 times (as an example), we can throw an appropriate error to the user.

Thanks for explaining this. It does make perfect sense, but maybe it's slightly better to provide consumers of the library with a more a flexibility in a form of 
 - a choice what to do next when a lock cannot be acquired even after, say, 5 seconds
 - an option to set the max number of attempts. 

In my opinion it's slightly better because different apps might need different backoff/retry strategies so that it's better to be able to control this behaviour from the caller side.

Reply:
The backoff / retry strategies can be implemented very easily using the functions provided by the library. I think the library should continue to have clean function signatures that are expected from a locking library.

However, if you may propose a different function signature for `acquireLock`, that allows for your requested changes and keeps the signature of the function "clean" :)

Reply:
Also agree with you, I was just trying to say that if you consider introducing a `maxAttempt` parameter, then it's better to expose via a function signature rather than hard code some value internally.